### PR TITLE
Pr/refactor tests

### PR DESCRIPTION
**Pull request by lubbo**
I know there are many changes, but please take this pull into account to cleanup Request/Response interface, remove redundant code and have a working test case.
I tried to keep ES retro compatibility, but I can't test ES below 7.0, so probably something should be fixed.
Feel free to discuss about it

Thanks!

### CODE REVIEW

File: Sources/ElasticSwift/ElasticSwift.swift [hunk 1]
```diff
@@ -16,24 +16,28 @@ public class RestClient: ESClient {
         self.init(settings: Settings.default)
     }
     
-    public func makeGet<T: Codable>() -> GetRequestBuilder<T> {
-        return GetRequestBuilder<T>(withClient: self)
+    public func makeGet<T: Codable>(fromIndex index: String, id: String) -> GetRequestBuilder<T> {
```
Comment (prafsoni) [inline Sources/ElasticSwift/ElasticSwift.swift:19]: `ElasticSwift` strictly follows builders pattern. I see the convenience of the proposed change however I see these changes as overload and not as a replacement.

File: Sources/ElasticSwift/Serialization/Serialization.swift [hunk 2]
```diff
@@ -8,20 +8,25 @@
 
 import Foundation
 
-class Serializers {
+public protocol Serializer {
```
Comment (prafsoni) [inline Sources/ElasticSwift/Serialization/Serialization.swift:11]: Great work! this change is awesome.

File: Sources/ElasticSwift/Requests/DeleteRequest.swift [hunk 3]
```diff
@@ -41,113 +45,57 @@ public class DeleteRequestBuilder: RequestBuilder {
         return self
     }
     
-    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> Void) -> Self {
+    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> ()) -> Self {
```
Comment (prafsoni) [inline Sources/ElasticSwift/Requests/DeleteRequest.swift:48]: Though, `Void` is a type alias for `()` we should use `Void` explicitly where the function should/does not return.

**Review by prafsoni — CHANGES_REQUESTED** [review]
Overall, I liked the changes proposed in the request. However, there a few minor changes that are needed. I have reflected them in comments.
Also, please make sure the PR doesn't have merge conflicts and is always submitted towards `dev`. I apologize for the inconvenience over branching as I only pointed towards `query-builders` branch but did make it clear to create pull request over dev branch.
To avoid these kinds of issues I am working on contributions guidelines document hoping to get it out soon.

**Comment by prafsoni** [comment]
@lubbo I would appreciate if you could provide feedback on `v1.0.0-alpha.5`. Also, I am looking forward to this PR with changes. cheers!

**Comment by lubbo** [comment]
Sure! This week I’m a bit busy but I’ll try my best

**Pull request closed without merging**